streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/ServiceConfigurationReader.java

```diff
@@ -0,0 +1,99 @@
+
+import com.google.common.collect.Lists;
+import com.hortonworks.streamline.common.QueryParam;
+import com.hortonworks.streamline.streams.catalog.Cluster;
+import com.hortonworks.streamline.streams.catalog.Namespace;
+import com.hortonworks.streamline.streams.catalog.NamespaceServiceClusterMapping;
+import com.hortonworks.streamline.streams.catalog.ServiceConfiguration;
+import com.hortonworks.streamline.streams.cluster.service.EnvironmentService;
+import com.hortonworks.streamline.streams.storm.common.ServiceConfigurationReadable;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ServiceConfigurationReader implements ServiceConfigurationReadable {
+
+    private final EnvironmentService environmentService;
+    private final long namespaceId;
+
+    public ServiceConfigurationReader(EnvironmentService environmentService, long namespaceId) {
+        this.environmentService = environmentService;
+        this.namespaceId = namespaceId;
+    }
+
+    @Override
+    public Map<Long, Map<String, String>> readAllClusters(String serviceName) {
+        Namespace namespace = environmentService.getNamespace(namespaceId);
+
+        if (namespace == null) {
+            throw new IllegalArgumentException("Namespace " + namespaceId + " doesn't exist.");
+        }
+
+        Long namespaceId = namespace.getId();
+
+        Collection<NamespaceServiceClusterMapping> mappings = environmentService.listServiceClusterMapping(namespaceId, serviceName);
+
+        List<Long> clusters = mappings.stream()
+                .map(mapping -> mapping.getClusterId())
+                .collect(Collectors.toList());
+
+        Map<Long, Map<String, String>> retMap = new HashMap<>();
+        clusters.forEach(c -> {
+            Map<String, String> flattenConfig = read(c, serviceName);
+            retMap.put(c, flattenConfig);
+        });
+
+        return retMap;
+    }
+
+    @Override
+    public Map<String, String> read(Long clusterId, String serviceName) {
+        Collection<NamespaceServiceClusterMapping> mappings = environmentService.listServiceClusterMapping(namespaceId, serviceName);
+        boolean associated = mappings.stream().anyMatch(map -> map.getClusterId().equals(clusterId));
+        if (!associated) {
+            return Collections.emptyMap();
+        }
+
+        Long serviceId = environmentService.getServiceIdByName(clusterId, serviceName);
+        if (serviceId == null) {
+            throw new IllegalStateException("Cluster " + clusterId + " is associated to the service " + serviceName +
+                " for namespace " + namespaceId + ", but actual service doesn't exist.");
+        }
+
+        Collection<ServiceConfiguration> serviceConfigurations = environmentService.listServiceConfigurations(serviceId);
```

Comment:
For hdfs in my test env, I see the following 4 configurations viz. hadoop-policy, core-site, hdfs-site and hadoop-env.

core-site and hdfs-site are fine, but I am not sure about the other two and if adding them will cause any un-intended effect. These key values finally gets inserted into a hadoop Configuration object. (Configuration for hdfs, HbaseConfiguration for hbase etc).